fixed: Image does not zoom where you click for desktop and web

metehanozyurt opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
For desktop and web versions big images can zoom clicked area problem solved.
### Fixed Issues
<!---
Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing.
Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the issue; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/8119

### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
1. Open Expensify on desktop app or web
2. Send horizontally and vertically big dimension image (for example : testFile3) 
3. Click image corners and center
4. Notice that it is zooming clicked position.
5. Send horizontally big dimension image (for example : testFile4) 
6. Click image corners and center
7. Notice that it is zooming clicked position.
8. Send vertically big dimension image (for example : testFile1) 
9. Click image corners and center
10. Notice that it is zooming clicked position.
11. Send horizontally and vertically small dimension image (for example : testFile2)
12. Click image corners and center
13. Notice that it is zooming image and fit screen keeping ratio
14. Click any dimension image on ios and android native apps double top first then pinch 
15. Notice that image is zoom to center after that pan should work
16. For mWeb pinch and double click not work. For ios pinch zooming modal only.

Test Files:
[testFile1](https://user-images.githubusercontent.com/6093207/165643550-104451ce-f722-4629-b9b0-d7efd6c30b1b.jpeg)
[testFile2](https://user-images.githubusercontent.com/6093207/165643564-1be64116-eafe-4086-b236-08c9c58e4073.jpeg)
[testFile3](https://user-images.githubusercontent.com/6093207/165643581-5673760a-0b67-4bd1-96c0-7c9f36a9d977.jpeg)
[testFile4](https://upload.wikimedia.org/wikipedia/commons/6/6c/Herjangsfjorden_%26_Ofotfjorden%2C_wide%2C_2009_09.jpg)

- [x] Verify that no errors appear in the JS console

### PR Review Checklist
<!--
This is a checklist for PR authors & reviewers. Please make sure to complete all tasks and check them off once you do, or else Expensify has the right not to merge your PR!
-->
#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.


<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Open Expensify on desktop app or web
2. Send horizontally and vertically big dimension image (for example : testFile3) 
3. Click image corners and center
4. Notice that it is zooming clicked position.
5. Send horizontally big dimension image (for example : testFile4) 
6. Click image corners and center
7. Notice that it is zooming clicked position.
8. Send vertically big dimension image (for example : testFile1) 
9. Click image corners and center
10. Notice that it is zooming clicked position.
11. Send horizontally and vertically small dimension image (for example : testFile2)
12. Click image corners and center
13. Notice that it is zooming image and fit screen keeping ratio
14. Click any dimension image on ios and android native apps double top first then pinch 
15. Notice that image is zoom to center after that pan should work
16. For mWeb pinch and double click not work. For ios pinch zooming modal only.

Test Files:
[testFile1](https://user-images.githubusercontent.com/6093207/165643550-104451ce-f722-4629-b9b0-d7efd6c30b1b.jpeg)
[testFile2](https://user-images.githubusercontent.com/6093207/165643564-1be64116-eafe-4086-b236-08c9c58e4073.jpeg)
[testFile3](https://user-images.githubusercontent.com/6093207/165643581-5673760a-0b67-4bd1-96c0-7c9f36a9d977.jpeg)
[testFile4](https://upload.wikimedia.org/wikipedia/commons/6/6c/Herjangsfjorden_%26_Ofotfjorden%2C_wide%2C_2009_09.jpg)


- [ ] Verify that no errors appear in the JS console

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
<!-- Insert screenshots of your changes on the web platform-->

https://user-images.githubusercontent.com/6093207/168396130-e1290b8c-ac4d-461c-86a1-0891f915043d.mp4

corner click video:


https://user-images.githubusercontent.com/6093207/166674708-0eca77ed-e0f7-4f9d-83a6-75481557d86a.mp4



#### Mobile Web
<!-- Insert screenshots of your changes on the web platform (from a mobile browser)-->

https://user-images.githubusercontent.com/6093207/165642136-cbfb7bde-4893-470a-bc2c-cdf2344d0a71.mp4


https://user-images.githubusercontent.com/6093207/165643036-c52f159f-fcf2-44fc-b1cd-ab231a23059d.mp4



#### Desktop
<!-- Insert screenshots of your changes on the desktop platform-->


https://user-images.githubusercontent.com/6093207/168415303-41aee154-4b2a-4a15-9ee8-6dd31b363ebb.mp4


corner click video:

https://user-images.githubusercontent.com/6093207/166674627-4ba29e00-4343-4b6c-824e-b4f06cae0834.mp4



#### iOS
<!-- Insert screenshots of your changes on the iOS platform-->

https://user-images.githubusercontent.com/6093207/165642141-b376dcbd-c093-4f48-91de-16bd3cfa114f.mp4


#### Android
<!-- Insert screenshots of your changes on the Android platform-->

https://user-images.githubusercontent.com/6093207/165642853-516585e9-d0bc-4a5e-a315-53ec3998f1dd.mp4

metehanozyurt committed:
fixed: Image does not zoom where you click for desktop and web

metehanozyurt commented on src/components/ImageView/index.js:
```diff
@@ -98,17 +94,22 @@ class ImageView extends PureComponent {
         }
 
-        if (this.state.isZoomed && this.state.isDragging && this.state.isMouseDown) {
+        if (this.isZoomed && this.state.isDragging && this.state.isMouseDown) {
```
We need to use it state.isZoomed to detect the change in rendering. I set the class level because when the state changes, the information of the clicked point is incorrect on containerPress . We hold state change , make calculations then set isZoomed state. 
I know it looks bad. But if we let the state change, we were losing the clicked coordinates.

metehanozyurt commented on src/components/ImageView/index.js:
```diff
@@ -129,55 +137,20 @@ class ImageView extends PureComponent {
-            imageRight: imgRight,
-            imageBottom: imgBottom,
-        });
+            zoomScale: containerHeight ? newZoomScale : prevState.zoomScale,
```
I'm a little obsessed with this comment (https://github.com/Expensify/App/issues/8119#issuecomment-1101107165) and i found the reason.
Usually setImageRegion calls first and we dont need to set zoomScale we just set the imgWidth and imgHeight. But sometimes the call order can be change and setImageRegion function calls last so we have last chance to set zoomScale.  Here video and picture:

https://user-images.githubusercontent.com/6093207/166516719-e3606a29-3457-427d-8a30-12af61d65c9b.mov



![WhatsApp Image 2022-04-09 at 14 41 10](https://user-images.githubusercontent.com/6093207/162572604-0be6564e-0d63-489f-8bd9-c484cdf5366f.jpeg)

chiragsalian replied:
i'm not sure how to reproduce but i don't mind keeping this. But let's add a comment explaining why its needed so that someone in future does not easily remove this by accident.

metehanozyurt committed:
Changed comments , added meaningful variable names

metehanozyurt committed:
Merge branch 'main' of https://github.com/Expensify/App into issue-8119

metehanozyurt committed:
Merge branch 'main' of https://github.com/Expensify/App into issue-8119

mananjadhav commented:
@metehanozyurt Ping me once you need me to take a look again.

Few more comments:
1. Can we add some comment for the updated logic. I'll also try to get more 👀 on the removed code.
2. Can you update the **Tests** section. I use this section to mention all the different tests that are performed. In your case, may be included small thumbnails, pinch works, etc.
3. If you could add screenshots for the scaling at the corners of the image. 

Once all comments are resolved, I'll take a look again and test it locally at my end.

metehanozyurt committed:
updated logic comments added

metehanozyurt commented on src/components/ImageView/index.js:
```diff
@@ -129,55 +137,20 @@ class ImageView extends PureComponent {
-
-        // return if image not loaded yet
-        if (imageHeight <= 0 || containerHeight <= 0) {
+        if (imageHeight <= 0) {
```
As I mentioned in the comment below (https://github.com/Expensify/App/pull/8809#discussion_r864069343) , it can sometimes be called last. therefore it is still necessary for the calculations we use below.

metehanozyurt committed:
correcting a typo

metehanozyurt commented:
Thank you @mananjadhav comments. 
I added comment on updated logic in my last commit. Fix my comments.
I have added detailed test steps. I hope I haven't added too much detail.
I also added videos of clicking the corners for the web and desktop versions.

I think I'm ready for you 😇.  Can you please take a look, thank you.

If you'll excuse me, I have a question. The PR I prepared solves the problem mentioned in the issue (https://github.com/Expensify/App/issues/8119). But there was no such support for mWeb before. For this reason, it is the principle of working as it was before. Is this a problem?

mananjadhav commented:
> I think I'm ready for you 😇. Can you please take a look, thank you.

Okay I'll go through this and get back.

> But there was no such support for mWeb before. For this reason, it is the principle of working as it was before. Is this a problem

IMO, it shouldn't be a problem. @chiragsalian can comment once. But quick question, are you aware of why isn't it working? Lack of platform support? Incorrect code, etc.?

chiragsalian commented:
Yeah its not a problem me too if you don't focus on mWeb for this. If there is a simple fix for mWeb that would be cool but you don't need to focus on it and we can open up another issue for mWeb if we feel its worth it.

chiragsalian commented:
Any update here. @mananjadhav, i believe its waiting for a review?

mananjadhav commented:
Will test tonight

mananjadhav commented on src/components/ImageView/index.js:
```diff
@@ -98,17 +94,26 @@ class ImageView extends PureComponent {
-            this.scrollableRef.scrollLeft = sX * this.state.zoomScale;
+
+            // We divide the clicked positions by the zoomScale.
+            // We need pixel coordinates.
```
```suggestion
            // Divide the clicked position by the zoomScale to get the pixel coordinates
```

mananjadhav commented:
PR Reviewed. @chiragsalian Looks good to me. Thanks, @metehanozyurt for the patience. It was a big change and I needed to be sure that it doesn't break anything.

1 minor comment and please merge the latest `main` branch

<details>
<summary>
#### PR Reviewer Checklist
</summary>
<p>

- [ ] I verified the PR has a small number of commits behind `main`
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the testing environment is mentioned in the test steps
- [x] I verified testing steps cover success & fail scenarios (if applicable)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors related to changes in this PR
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified comments were added when the code was not self explanatory
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files (if applicable)
    - [x] I verified proper naming convention for platform-specific files was followed (if applicable)
    - [ ] I verified [style guidelines](https://github.com/Expensify/App/blob/main/STYLING.md) were followed
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components are not impacted by changes in this PR (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified the UI performance was not affected (the performance is the same than `main` branch)
- [ ] If a new component is created I verified that a similar component doesn't exist in the codebase

</p>
</summary>
</details>

chiragsalian commented on src/components/ImageView/index.js:
```diff
@@ -117,67 +122,41 @@ class ImageView extends PureComponent {
-            isMouseDown: false,
-        }));
+        // We hold for set isZoomed state if true
+        // Because when set isZoomed state can't getting actual position user clicked
```
I don't follow your comments here. Can you rephrase? I think you are trying to say `onContainerPress` calculates where to scroll so we zoom in over here and not here, is that correct?
If so how does this rephrase sound?
```
// We won't set isZoomed true in the block below and instead set it in onContainerPress as that method calculates the scroll positions after zooming.
```

chiragsalian commented on src/components/ImageView/index.js:
```diff
@@ -117,67 +122,41 @@ class ImageView extends PureComponent {
+        // We hold for set isZoomed state if true
+        // Because when set isZoomed state can't getting actual position user clicked
+        this.isZoomed = !this.state.isZoomed;
+        if (this.isZoomed === false) {
```
I'm a little confused why this logic is in `onContainerPressOut` and why `this.isZoomed` is needed.
I feel like this can be simplified if it lived in `onContainerPress`, then we wouldn't need this.isZoomed or even need to do multiple setStates.

So wouldn't this  `onContainerPress` code work? (untested)
```js
onContainerPress(e) {
    let scrollX;
    let scrollY;
    if (!this.state.isZoomed && !this.state.isDragging) {
        const {offsetX, offsetY} = e.nativeEvent;

        // We divide the clicked positions by the zoomScale.
        // We need pixel coordinates.
        const delta = this.getScrollOffset(offsetX / this.state.zoomScale, offsetY / this.state.zoomScale);
        scrollX = delta.offsetX;
        scrollY = delta.offsetY;
    }

    if (this.state.isZoomed && this.state.isDragging && this.state.isMouseDown) {
        this.setState({isDragging: false, isMouseDown: false});
    } else {
        // We first zoom and once its done if we are zooming in then we scroll to the location the user clicked.
        this.setState(prevState => ({
            isZoomed: !prevState.isZoomed,
            isMouseDown: false,
        }), () => {
            this.scrollableRef.scrollTop = scrollY;
            this.scrollableRef.scrollLeft = scrollX;
        });
    }
}
```

metehanozyurt replied:
It worked perfectly. I wish I could think of that. Thank you so much. Here is the test video. I will use this code and re-shoot videos for web and desktop.

https://user-images.githubusercontent.com/6093207/168396130-e1290b8c-ac4d-461c-86a1-0891f915043d.mp4

Thanks to this update, we can delete these lines.

https://github.com/Expensify/App/blob/9a8291ba97c381b651b5344d6fa471c51ae70336/src/components/ImageView/index.js#L26

https://github.com/Expensify/App/blob/9a8291ba97c381b651b5344d6fa471c51ae70336/src/components/ImageView/index.js#L115-L124

https://github.com/Expensify/App/blob/9a8291ba97c381b651b5344d6fa471c51ae70336/src/components/ImageView/index.js#L287

chiragsalian commented on src/components/ImageView/index.js:
```diff
@@ -187,31 +166,23 @@ class ImageView extends PureComponent {
-        if (y < this.state.imageTop) {
-            sy = 0;
+        // Container size bigger than clicked position offset
+        if (x <= (this.state.containerWidth / 2)) {
```
remove unnecessary brackets, it should be like so
```
x <= this.state.containerWidth / 2
```

chiragsalian reviewed (changes requested):
Works well, left a few comments. Additionally in your test steps you mention,
> 14-) Click any dimension image on ios and android native apps notice that pinch and pan works. For Android need too double top image first then pinch works like on test video.
15-) for mWeb pinch and double click not work. For ios pinch zooming modal only.


For iOS native i also needed to double click to zoom into image, and it only can zoom to center and not a specific area. While i think thats fine i wasn't sure if you test steps mentioned the same for iOS or if the expectation was different. Can you make it more clear?

<details>
  <summary>PR reviewer checklist</summary>
  
  - [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [ ] Android / native
    - [x] iOS / Safari
    - [ ] Android / Chrome
    - [x] MacOS / Chrome
    - [ ] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn’t already exist
    - [x] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
</details>

metehanozyurt committed:
unnecesary function removed, better comments added, code has been improved.

metehanozyurt committed:
Merge branch 'main' of https://github.com/Expensify/App into issue-8119

metehanozyurt committed:
unnecesary condition removed.

metehanozyurt commented:
Thank you for your valuable comments and patience @mananjadhav and @chiragsalian. I will study harder to give better explanations.

You are right about this line. I've been paranoid. I took it back as it was an unnecessary condition.
```js 
zoomScale: containerHeight ? newZoomScale : prevState.zoomScale,
```

I shot new videos for desktop and web. It works as it should, after new changes.
I updated the test steps. I tried to be a little more understandable.

I hope you like it, thank you very much again 🙏 .

mananjadhav commented:
Thanks for the changes @metehanozyurt. Appreciate your patience. I will test this again tonight.

Thanks, @chiragsalian for the suggestions.

mananjadhav reviewed (approved):
LGTM. Thanks, @metehanozyurt for the changes.

@chiragsalian All yours.

🎀 👀 🎀 C+ reviewed

chiragsalian reviewed (approved):
Nice, this looks much cleaner. Thank you for the changes @metehanozyurt.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [ ] Android / native
    - [x] iOS / Safari
    - [ ] Android / Chrome
    - [x] MacOS / Chrome
    - [ ] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn’t already exist
    - [x] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

metehanozyurt commented:
Thank you so much for helps and patience @chiragsalian , @mananjadhav 🙏 .

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/2340088077) to staging by @chiragsalian in version: 1.1.62-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
